### PR TITLE
add passage highlighting

### PR DESCRIPTION
**Pull request by rhagarty**
if doing a 'passage' query, highlight the passages in the returned list of matches

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -84,10 +68,118 @@ const sortTypes = [
...
+      sentimentLabel: dataItem.enriched_text.sentiment.document.label,
+      hasPassage: false,
+      passageField: '',
+      passageScore: '0'
+    };
+
+    if (passages.results) {
+      for (var i=0; i<passages.results.length; i++) {
+        var res = passages.results[i];
+        // if (i == 1) console.log('compate passage id: ' + res.document_id + ' to: ' + dataItem.id);
```
Comment (markstur) [inline lib/utils.js]: should clean out some of these commented things.  And "compate"?

File: lib/utils.js [hunk 2]
```diff
@@ -84,10 +68,118 @@ const sortTypes = [
...
   { key: BY_WORST,   value: BY_WORST_QUERY,   text: 'Lowest Rated'} ];
+  
+
+const objectWithoutProperties = (object, properties) => {
+  'use strict';
+
+  var obj = {};
+  var keys = Object.keys(object);
+  keys.forEach(key => {
+    if (!~properties.indexOf(key)) {
```
Comment (markstur) [inline lib/utils.js]: i think the ~ trick is not good for code readers.

File: lib/utils.js [hunk 3]
```diff
@@ -84,10 +68,118 @@ const sortTypes = [
...
+        // if (i == 1) console.log('compate passage id: ' + res.document_id + ' to: ' + dataItem.id);
+        if (res.document_id === dataItem.id) {
+          // console.log('FOUND Passage: ' + res.document_id);
+          newResult.hasPassage = true;
+          newResult.passageStart = res.start_offset;
+          newResult.passageEnd = res.end_offset;
+          newResult.passageField = res.field;
+          newResult.passageScore = res.passage_score;
+          // console.log("RESULT: ");
+          // const util = require('util');
```
Comment (markstur) [inline lib/utils.js]: like here, I understand if you want to keep some debug stuff, but did you want to keep the commented require?   Well, maybe you did.

File: src/main.js [hunk 4]
```diff
@@ -1001,4 +1001,4 @@ Main.propTypes = {
   error: PropTypes.object
 };
 
-module.exports = Main;
+module.exports = Main;
```
Comment (markstur) [inline src/main.js]: nit: missing EOL

File: lib/utils.js (hunk 2 above)
Comment (rhagarty) [reply lib/utils.js]: agree - will switch to '< 0'

File: lib/utils.js (hunk 1 above)
Comment (rhagarty) [reply lib/utils.js]: done

File: lib/utils.js (hunk 3 above)
Comment (rhagarty) [reply lib/utils.js]: yeah, like to have the require here since it's only needed for this one console.log call

File: src/main.js (hunk 4 above)
Comment (rhagarty) [reply src/main.js]: done

**Pull request merged**